sdk/iot/provisioning/src/az_iot_provisioning_client.c

```diff
@@ -151,23 +151,96 @@ AZ_NODISCARD az_result az_iot_provisioning_client_register_get_subscribe_topic_f
 
   az_span mqtt_topic_filter_span
       = az_span_init((uint8_t*)mqtt_topic_filter, (int32_t)mqtt_topic_filter_size);
-  int32_t required_length = 1 + az_span_size(str_dps_registrations_res);
+  int32_t required_size = 1 + 1 + az_span_size(str_dps_registrations_res);
 
-  AZ_RETURN_IF_NOT_ENOUGH_SIZE(mqtt_topic_filter_span, required_length);
+  AZ_RETURN_IF_NOT_ENOUGH_SIZE(mqtt_topic_filter_span, required_size);
 
   az_span remainder = az_span_copy(mqtt_topic_filter_span, str_dps_registrations_res);
   remainder = az_span_copy_u8(remainder, '#');
   remainder = az_span_copy_u8(remainder, '\0');
 
   if (out_mqtt_topic_filter_length)
   {
-    *out_mqtt_topic_filter_length = (size_t)required_length;
+    *out_mqtt_topic_filter_length = (size_t)required_size - 1;
+  }
+
+  return AZ_OK;
+}
+
+// $dps/registrations/PUT/iotdps-register/?$rid=%s
+AZ_NODISCARD az_result az_iot_provisioning_client_register_get_publish_topic(
+    az_iot_provisioning_client const* client,
+    char* mqtt_topic,
+    size_t mqtt_topic_size,
+    size_t* out_mqtt_topic_length)
+{
+  (void)client;
+
+  AZ_PRECONDITION_NOT_NULL(client);
+  AZ_PRECONDITION_NOT_NULL(mqtt_topic);
+  AZ_PRECONDITION(mqtt_topic_size > 0);
+
+  az_span mqtt_topic_span = az_span_init((uint8_t*)mqtt_topic, (int32_t)mqtt_topic_size);
+  az_span str_dps_registrations = _az_iot_provisioning_get_str_dps_registrations();
+
+  int32_t required_size
+      = 1 + az_span_size(str_dps_registrations) + az_span_size(str_put_iotdps_register);
+
+  AZ_RETURN_IF_NOT_ENOUGH_SIZE(mqtt_topic_span, required_size);
+
+  az_span remainder = az_span_copy(mqtt_topic_span, str_dps_registrations);
+  remainder = az_span_copy(remainder, str_put_iotdps_register);
+  remainder = az_span_copy_u8(remainder, '\0');
+
+  if (out_mqtt_topic_length)
+  {
+    *out_mqtt_topic_length = (size_t)required_size - 1;
+  }
+
+  return AZ_OK;
+}
+
+// Topic: $dps/registrations/GET/iotdps-get-operationstatus/?$rid=%s&operationId=%s
+AZ_NODISCARD az_result az_iot_provisioning_client_query_status_get_publish_topic(
+    az_iot_provisioning_client const* client,
+    az_iot_provisioning_client_register_response const* register_response,
+    char* mqtt_topic,
+    size_t mqtt_topic_size,
+    size_t* out_mqtt_topic_length)
+{
+  (void)client;
```

Comment:
Do we intend to use the `client` in the future?

Reply:
Not that I know of.